### PR TITLE
[WIP] Add sprint

### PR DESCRIPTION
**Pull request by deadcutlass2**
```SUMMARY: Balance/Features "Adds sprint movement mode"```

#### Purpose of change
The survivor was rather slow before, clocking his blazing jog of 2 m/s for less than 2 minutes. This seems pathetic by real worlds standards. Added a faster sprint and rebalanced his jog. Made carry weight mean something in stamina regen.

#### Describe the solution
Added an actually fast sprint that reduces move cost by a factor of 8, instead of runnings factor of 2.
Stamina cost of sprint is 16x base
Stamina cost of running reduces to 3x base from 7x base
Base stamina cost reduced to 10 from 15
Carry weight penalty is applied to stamina burn.
1% increased stamina cost for each % of your max weight you are carrying, which brings sprinting cost up to ~320 per turn and running up to ~60 per turn, compared to runnings previous of ~70 per turn regardless of if the survivor was fully equipped or stark naked.

#### Describe alternatives you've considered
Increasing base character movespeed to bring it up to real world values, thus making running somewhat faster than the average jogger. Though this messes with a lot of things.

#### Additional context
https://spotterup.com/the-soldiers-ideal-speed/
"When it comes to sprint standards, with or without 16kg of equipment, 60m in 8sec and 100m in 12sec is both reasonable and essential in combat. Faster is always better, as is the ability to repeat."

### CODE REVIEW

**Comment by Inglonias** [comment]
While this is nice in theory, we already have three different walking speeds. This is starting to get a little excessive.

**Comment by TechyBen** [comment]
Is the regent debuff on weight while resting or while walking? As it's strange to apply a stamina regen restriction while standing still, even if "carrying" heavy loads. But while moving, would of cause limit stamina regen.

File: src/action.cpp [hunk 1]
```diff
@@ -683,6 +685,10 @@ action_id handle_action_menu()
         action_weightings[ACTION_TOGGLE_RUN] = 300;
     }
     // If we're already crouching, make it simple to toggle crouching to off.
```
Comment (BevapDin) [inline src/action.cpp]: You presumably copied that comment from below. It makes little sense here (it's talking about crouching).

**Comment by deadcutlass2** [comment]
First to TechyBen, the stamina burn only applies on movement, not on standing around. Base stamina burn in PR is 10, which with the weight penalty can double up to 20 without going over your weight capacity. Base stamina regen is 20, meaning that an unencumbered survivor can carry their maximum load at walking speeds without burning stamina. 

To BevapDin, yes.

Next to Kevin, I would like some hard numbers on how large each tile is supposed to be relative to our movement system. Based on the mention of 40y/5s being 5x nominal walking speed, I assume now that it is supposed to be each tile is 1.6 yards or 1.5 meters, but I'd like to get that from you before adjusting.

As for balance, The first thing I can think of is adjusting the base zombie speed to be just under a player's run. This would put walking as your out of combat movement speed, running for in combat, and sprinting to quickly leave a situation.

Another option is as you suggested to cause carrying weight to impact movement speed as well. I am unsure by how much to reduce it by. My first impression is to halve movement speed as capacity approaches maximum. However, the source that I listed mentions that sprinting should be able to be accomplished even with a 16 kg load, so maybe start the penalties towards movement speed at half of maximum weight capacity?

Also optionally is to drop sprinting and open a different PR with just changes to stamina burn rates based on carry capacity.

**Comment by kevingranade** [comment]
> I would like some hard numbers on how large each tile is supposed to be relative to our movement system.

Me too, but it doesn't exist, also it doesn't matter. What is important is relative speeds, not absolute.

> This would put walking as your out of combat movement speed, running for in combat, and sprinting to quickly leave a situation.

This doesn't address balance at all, the question is under what circumstances the player *can* escape combat, and as far as I can tell, your changes mean the player can escape any time they have stamina reserves.
Making sprinting work would not be a matter of making a few tweaks, it would most likely require overhauling a lot of game balance, and I just have to ask, what does this buy us? At the end of the day, this kind of thing MUST be zero-sum, where this option doesn't make survival easier.

> Also optionally is to drop sprinting and open a different PR with just changes to stamina burn rates based on carry capacity.

This is a really good idea, because this is a much easier thing to achieve.

**Comment by Zombiemanpig** [comment]
> A lot of this change is built on a shaky premise that one tile = 1 meter, which from what I can see invalidates everything except applying carry weight speed to stamina burn.
> 
> A move cost increase of 8x compared to walk speed is simply ludicrous, a quick check of records turns up NFL sprinting speed records around the 40yards/5seconds mark, which represents about a 5x increase compared to nominal walking speed. Yes there are higher sprinting records, but your character is not an Olympic athlete in peak conditioning a carefully prepared track.
> Even that needs to decay based on carried load and encumbrance, there's a reason the current run speed is set to be so conservative, it provides an achievable speed boost without complicating things.
> Speaking of complications, an all-out sprint should impair the player's ability to change direction and require acceleration to reach top speed, maintaining a sprinting speed while making 90 degree turns is not possible.
> 
> The other thing you've neglected is balance, we cannot give the player drastic new move modes without adjusting the game to match or at least consider whether it breaks it (hint, it does).
> 
> Finally, your code changes don't do what you think they do, if you can resolve the balance issues, you need to do some testing to verify that it works before submitting a change.

Maybe cause you are not a runner, but you don't start your sprint at 100 percent of your speed, it takes a solid second or 2 to even reach your max speed so using NFL sprint times is very misleading and not realistic.

**Comment by anothersimulacrum** [comment]
This is in that quote block.

> Speaking of complications, an all-out sprint should impair the player's ability to change direction and require acceleration to reach top speed

**Comment by Zombiemanpig** [comment]
> This is in that quote block.
> 
> > Speaking of complications, an all-out sprint should impair the player's ability to change direction and require acceleration to reach top speed

And in the same quote he says 

>A move cost increase of 8x compared to walk speed is simply ludicrous, a quick check of records >turns up NFL sprinting speed records around the 40yards/5seconds mark, which represents about a >5x increase compared to nominal walking speed.

Yet they do move at about 8x the speed and using the NFL sprint times has the same flaw he just mentioned.

Also he says it would be hard to add things in like turning (I agree) but acceleration can easily be added as an inbetween 1 or 2 turn debuff / buff that increases or decreases the movement speed.

**Pull request closed without merging**